### PR TITLE
feat: #119645 Add detection for Samsung internet browser

### PR DESCRIPTION
**Pull request by KSkrede**
## Prerequisites

Please make sure you can check the following boxes:

-   [x] My code follows the code style of this project
-   [x] All new and existing tests passed
-   [x] Fix any eslint/prettier warnings/errors: npm run lint
-   [x] Remove unnecessary console.log (use logInfo logWarn if it's needed)
-   [x] Are variable and function names properly describing what it does?

### Contribution Type

What types of changes does your code introduce? Put an `x` in all the boxes that apply:

-   [ ] Bug fix (non-breaking change which fixes an issue, please reference the issue id)
-   [x] New feature (non-breaking change which adds functionality, make sure to open an associated issue first)
-   [ ] Breaking change (fix or feature that would cause existing functionality to change)
-   [ ] I have updated the documentation accordingly
-   [ ] I have added tests to cover my changes

### Description

Added detection of Samsung Internet browser.
 
Justification for PR: https://dev.azure.com/EquinorASA/Echo/_sprints/backlog/Discovery%20one/Echo/Increment%202%20-%202023Q2/Sprint%203%20(summer)?workitem=119645

https://gs.statcounter.com/browser-market-share/mobile/norway

### CODE REVIEW

**Review by OveBDev — APPROVED** [review]
Looks good, just add the us/bug id to the title: feat: #119645 Add...
This is a requirement in Equinor.

**Comment by KSkrede** [comment]
Thanks @OveBDev! Forgot to change PR to draft, as I was unsure if the us had to be EchoUtils specific or if it could come from another project. Does it look good now?

File: src/utils/deviceInfo/deepDeviceInfo.ts [hunk 1]
```diff
@@ -161,6 +165,19 @@ class DetailedDeviceInformationProvider {
         if (!model && !vendor) return '';
         return `${model} ${vendor}`;
     }
+
+    private static getDeviceType(dataOrigin: UAParser): DeviceType {
+        const deviceType = dataOrigin.getDevice().type;
+        if (deviceType) {
```
Comment (CsabaTheKiss) [inline src/utils/deviceInfo/deepDeviceInfo.ts:171]: Could remove nested if with early return: it makes the code more readable.
```ts
if(!deviceType) {
   return 'desktop';
}
// rest of the code...
if (deviceType.includes('mobile')) {
   return 'mobile';
}
...
```

File: src/utils/deviceInfo/tests/deepDeviceInfo.test.ts [hunk 2]
```diff
@@ -74,6 +74,16 @@ describe('A browser that doesnt support userAgentData', () => {
         expect(deviceInfo.deviceType).toBe('desktop');
         expect(deviceInfo.screenDimensions).toBeDefined();
     });
+
+    test('Simulated Samsung device running Samsung Internet', () => {
+        const mockedUserAgent =
```
Comment (CsabaTheKiss) [inline src/utils/deviceInfo/tests/deepDeviceInfo.test.ts:79]: Consistency: Could you add newlines, just to group the different parts of the test (in the similar manner as the other tests in this file)?

File: src/utils/deviceInfo/deepDeviceInfo.ts (hunk 1 above)
Comment (eiriknysted) [reply src/utils/deviceInfo/deepDeviceInfo.ts]: will do, thanks!

**Pull request merged**